core/src/cdn_url.rs

```diff
@@ -149,14 +166,17 @@ impl TryFrom<CdnUrlMessage> for MaybeExpiringUrls {
                         String::new()
                     };
 
-                    let mut expiry: i64 = expiry_str.parse()?;
-
-                    expiry -= 5 * 60; // seconds
-
-                    Ok(MaybeExpiringUrl(
-                        cdn_url.to_owned(),
-                        Some(Date::from_timestamp_ms(expiry * 1000)?),
-                    ))
+                    let expiry = match expiry_str.parse::<i64>() {
+                        Ok(exp) => {
+                            let exp = exp - CDN_URL_EXPIRY_MARGIN_SECONDS;
```

Comment:
Will probably go automatically when changing to `Duration` but let's use some sort of `checked` or `saturating` sub to prevent panics. Spotify can output weird stuff...